src/ora/src/Ora.cc

```diff
@@ -0,0 +1,276 @@
+        "your "
+        "queries and the subsequently generated answers. Give consent to use "
+        "the application by using the set_consent command.");
+
+    return;
+  }
+  logger_->info(utl::ORA, 101, "Sending POST request to {}", hostUrl);
+
+  std::stringstream jsonDataStream;
+  jsonDataStream << R"({
+        "query": ")"
+                 << query << R"(",
+        "chat_history": [],
+        "list_sources": )"
+                 << (sourceFlag_ ? "true" : "false") << R"(
+    })";
+
+  std::string jsonData = jsonDataStream.str();
+  std::string postResponse = sendPostRequest(hostUrl, jsonData);
+
+  if (postResponse.empty()) {
+    logger_->warn(utl::ORA, 102, "No response received from API.");
+    return;
+  }
+
+  try {
+    boost::json::value jsonObject = boost::json::parse(postResponse);
+
+    if (jsonObject.as_object().contains("response")
+        && jsonObject.at("response").is_string()) {
+      std::string response = jsonObject.at("response").as_string().c_str();
+
+      if (sourceFlag_ && jsonObject.as_object().contains("sources")
+          && jsonObject.at("sources").is_array()) {
+        std::string sources;
+        for (const auto& src : jsonObject.at("sources").as_array()) {
+          sources += src.as_string().c_str();
+          sources += "\n";
+        }
+        logger_->info(utl::ORA,
+                      103,
+                      "ORAssistant Response: \n\n{}\nSources:\n{}",
+                      response,
+                      sources);
+      } else {
+        logger_->info(
+            utl::ORA, 104, "ORAssistant Response: \n\n{}\n", response);
+      }
+    } else {
+      logger_->warn(utl::ORA,
+                    106,
+                    "API response does not contain 'response' field or it is "
+                    "not a string.");
+      logger_->warn(utl::ORA, 107, "API response: {}", postResponse);
+    }
+  } catch (const boost::json::system_error& e) {
+    logger_->warn(utl::ORA,
+                  105,
+                  "JSON Parsing Error: {}\nPlease check if you have access to "
+                  "ORAssistant's API.",
+                  e.what());
+  }
+}
+
+void Ora::setSourceFlag(bool sourceFlag)
+{
+  sourceFlag_ = sourceFlag;
+}
+
+void Ora::setBotHost(const char* host)
+{
+  hostUrl = host;
+  logger_->info(utl::ORA, 116, "Setting ORAssistant host to {}", hostUrl);
+
+  checkLocalDir();
+  std::string hostFilePath = localDirPath + "/orassistant_host.txt";
+  std::ofstream hostUrlFile(hostFilePath);
+
+  if (hostUrlFile.is_open()) {
+    hostUrlFile << hostUrl;
+    hostUrlFile.close();
+    logger_->info(utl::ORA,
+                  109,
+                  "ORAssistant host saved to "
+                  "~/.local/share/openroad/orassistant_host.txt");
+  } else {
+    logger_->warn(utl::ORA, 108, "Failed to write ORAssistant host to file.");
+  }
+}
+
+void Ora::setConsent(const char* consent)
```

Comment:
since you are checking for just a "y" for yes, I think you need to verify that this consent string matches your expectations.

Reply:
I'm checking this (along with the mode string) and printing appropriate error msgs in the `Ora.i` file.